src/hotspot/share/opto/superword.cpp

```diff
@@ -38,146 +38,51 @@
-  _early_return(true),                                      // analysis evaluations routine
-  _do_vector_loop(phase->C->do_vector_loop()),              // whether to do vectorization/simd style
+  _do_vector_loop(phase()->C->do_vector_loop()),            // whether to do vectorization/simd style
   _num_work_vecs(0),                                        // amount of vector work we have
   _num_reductions(0)                                        // amount of reduction work we have
 {
 }
 
-//------------------------------transform_loop---------------------------
-bool SuperWord::transform_loop(IdealLoopTree* lpt, bool do_optimization) {
-  assert(_phase->C->do_superword(), "SuperWord option should be enabled");
-  // SuperWord only works with power of two vector sizes.
-  int vector_width = Matcher::vector_width_in_bytes(T_BYTE);
-  if (vector_width < 2 || !is_power_of_2(vector_width)) {
-    return false;
-  }
-
-  assert(lpt->_head->is_CountedLoop(), "must be");
-  CountedLoopNode *cl = lpt->_head->as_CountedLoop();
-
-  if (!cl->is_valid_counted_loop(T_INT)) {
-    return false; // skip malformed counted loop
-  }
-
-  // Initialize simple data used by reduction marking early.
-  set_lpt(lpt);
-  set_lp(cl);
-  // For now, define one block which is the entire loop body.
-  set_bb(cl);
-
-  if (SuperWordReductions) {
-    mark_reductions();
-  }
-
-  // skip any loop that has not been assigned max unroll by analysis
-  if (do_optimization) {
-    if (SuperWordLoopUnrollAnalysis && cl->slp_max_unroll() == 0) {
-      return false;
-    }
-  }
-
-  // Check for no control flow in body (other than exit)
-  Node *cl_exit = cl->loopexit();
-  if (cl->is_main_loop() && (cl_exit->in(0) != lpt->_head)) {
-    #ifndef PRODUCT
-      if (is_trace_superword_precondition()) {
-        tty->print_cr("SuperWord::transform_loop: loop too complicated, cl_exit->in(0) != lpt->_head");
-        tty->print("cl_exit %d", cl_exit->_idx); cl_exit->dump();
-        tty->print("cl_exit->in(0) %d", cl_exit->in(0)->_idx); cl_exit->in(0)->dump();
-        tty->print("lpt->_head %d", lpt->_head->_idx); lpt->_head->dump();
-        lpt->dump_head();
-      }
-    #endif
-    return false;
-  }
-
-  // Make sure the are no extra control users of the loop backedge
-  if (cl->back_control()->outcnt() != 1) {
-    return false;
-  }
-
-  // Skip any loops already optimized by slp
-  if (cl->is_vectorized_loop()) {
-    return false;
-  }
-
-  if (cl->is_unroll_only()) {
-    return false;
-  }
-
-  if (cl->is_main_loop()) {
-    // Check for pre-loop ending with CountedLoopEnd(Bool(Cmp(x,Opaque1(limit))))
-    CountedLoopEndNode* pre_end = cl->find_pre_loop_end();
-    if (pre_end == nullptr) {
-      return false;
-    }
-    Node* pre_opaq1 = pre_end->limit();
-    if (pre_opaq1->Opcode() != Op_Opaque1) {
-      return false;
-    }
-    cl->set_pre_loop_end(pre_end);
-  }
-
-  init(); // initialize data structures
```

Comment:
Note: this is the end of the "preconditions", and we used to set `_early_exit = false` inside `init()`